enhancements/api-review/add-repositoryMirrors-spec.md

```diff
@@ -0,0 +1,175 @@
+---
+title: add-repositoryMirrors-spec
+authors:
+  - "@QiWang19"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-03-10
+last-updated: 2021-03-10
+status: implementable
+---
+
+# Add repositoryMirrors spec to cluster wide ImageContentSourcePolicy
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Today, the ImageContentSourcePolicy object sets up mirror with `mirror-by-digest-only` property set to true. Requires images be pulled by digest only. This enhancement plans to add new spec to ImageContentSourcePolicy so mirrors can be used without setting up default `mirror-by-digest-only = true`.
+
+## Motivation
+
+Today, the ImageContentSourcePolicy object sets up mirror configuration with `mirror-by-digest-only` property set to true, which leads to using mirrors only when images are referenced by digests. However, when working with disconnected environments, sometimes multiple ImageContentSourcePolicies are needed, some of them are used by apps/manifests that don't use digests when pulling the images.
+
+Adding the `repositoryMirrors` spec to ImageContentSourcePolicy to configure mirrors without setting `mirror-by-digest-only` will make it possible to pull images from mirror using tags without requiring image digest reference.
+
+### Goals
+
+- Enable the user to pull images through ImageContentSourcePolicy configured mirrors by tags.
+
+### Non-Goals
+
+## Proposal
+
+The Image API has an optional `MirrorByDigestOnly` filed with type `bool` to `Registry`.
+
+```go
+// Registry represents a registry.
+type Registry struct {
+  // ...
+
+	// If true, mirrors will only be used for digest pulls. Pulling images by
+	// tag can potentially yield different images, depending on which endpoint
+	// we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+	MirrorByDigestOnly bool `toml:"mirror-by-digest-only,omitempty"`
```

Comment:
Defaulting to the unsafe `false` is riskier than I'd like.  If this is a true boolean in `registries.conf` (I don't know), I'd prefer `AllowTags` or some such here to default safe.

Reply:
Isn't mirror by digest the default behavior today? If that is the case then then new API needs to be opt-in, as suggested with `AllowMirrorByTags`

Reply:
These lines of API definition is current `registries.conf`. Not the ImageCententSourcePolicy that this enhancement try to implement.

Reply:
> Isn't mirror by digest the default behavior today? If that is the case then then new API needs to be opt-in, as suggested with `AllowMirrorByTags`

I have updated this enhancement to add a new spec of `repositoryMirrros` per comment https://github.com/openshift/enhancements/pull/690#discussion_r632822088